### ISSUE TITLE
Wazuh stops when an invalid Syscollector configuration is applied

### ISSUE THREAD

**Issue by mauromalara** [wazuh/wazuh#18968]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.6.0-40600 | Syscollector | Agent | Packages | Windows-10-10.0.14393-SP0 |

While running the integration tests for `4.6.0-alpha1` we found that when using an invalid Syscollector configuration Wazuh does not start correctly. So, the tests fail because they expect some logs to appear, but they never do.

**Evidence**: [syscollector-agent-windows.zip](https://github.com/wazuh/wazuh/files/12581433/syscollector-agent-windows.zip)

**Note**: The `test_syscollector_scanning` test fails because of a known issue: https://github.com/wazuh/wazuh-qa/issues/4230

**Error preview**:
```
2023/09/11 12:45:21 wazuh-agent[3664] logging_helper.c:53 at loggingFunction(): CRITICAL: Error, id: dbEngine: Error deleting old db.
2023/09/11 12:45:21 wazuh-agent[3664] win_service.c:262 at OssecServiceCtrlHandler(): INFO: Received exit signal. Starting exit process.
2023/09/11 12:45:21 wazuh-agent[3664] win_service.c:268 at OssecServiceCtrlHandler(): INFO: Set pending exit signal.
2023/09/11 12:45:21 wazuh-agent[3664] win_service.c:278 at OssecServiceCtrlHandler(): INFO: Exit completed successfully.
2023/09/11 12:45:21 wazuh-agent[3664] cryptography.c:541 at verify_hash_and_pe_signature(): DEBUG: PE signature verification succeeded for C:\Windows\system32\mswsock.dll
2023/09/11 12:45:21 wazuh-agent[3664] dll_load_notify.c:101 at dll_notification(): DEBUG: The file 'C:\Windows\system32\mswsock.dll' is signed and its signature is valid.
```

**Comment by mauromalara** [comment]
Seems to be related with: https://github.com/wazuh/wazuh/issues/16080

**Comment by Dwordcito** [comment]
The issue is not reproducible in our environments, please help Gabriel to reproduce them.

**Comment by mauromalara** [comment]
## Update

- I tried to reproduce this in my environment with several configurations and use cases and it did not fail (as @GabrielEValenzuela mentioned before).
- I did execute the tests in Jenkins again to see if this is something that happens more than once, and it fails again. Same tests.
- I set an environment like the one in Jenkins to launch the tests with `pytests` locally and debug the code in order to find where the error/defect is
- It is likely a problem with the tests not with Wazuh itself.

**Comment by mauromalara** [comment]
## Conclusion

**Is a race condition produced by the tests.** 🔴

***

### Context/background:
- No related code to the error was modified in `4.6.0`
  - This was checked in conjunction with @GabrielEValenzuela 
- The test was no modified in `4.6.0`
- The test passed in `4.5.2`: https://github.com/wazuh/wazuh/issues/18622

### Manual tests

After running manual tests, the error cannot be reproduced, either by **[me](https://github.com/wazuh/wazuh/issues/18968#issuecomment-1720262544)** or **[the developer](https://github.com/wazuh/wazuh/issues/18968#issuecomment-1717884976)**.


### Automated tests execution

I executed the tests and **now the error appears**.

So, we can conclude that the error appears only when automatic tests are executed.

<details><summary>Execution details</summary></br>

- Manager system info
```shell
# cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.2 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.2 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy


# df -H --total
Filesystem       Size  Used Avail Use% Mounted on
/dev/root         32G  2.9G   29G  10% /
tmpfs            2.1G   82k  2.1G   1% /dev/shm
tmpfs            813M  840k  812M   1% /run
tmpfs            5.3M     0  5.3M   0% /run/lock
/dev/nvme0n1p15  110M  6.4M  104M   6% /boot/efi
tmpfs            407M  4.1k  407M   1% /run/user/0
total             35G  2.9G   32G   9% -


# lscpu
Architecture:            x86_64
  CPU op-mode(s):        32-bit, 64-bit
  Address sizes:         48 bits physical, 48 bits virtual
  Byte Order:            Little Endian
CPU(s):                  2
  On-line CPU(s) list:   0,1
Vendor ID:               AuthenticAMD
  Model name:            AMD EPYC 7571
    CPU family:          23
    Model:               1
    Thread(s) per core:  2
    Core(s) per socket:  1
    Socket(s):           1
    Stepping:            2
    BogoMIPS:            4399.99
    Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonsto
                         p_tsc cpuid extd_apicid tsc_known_freq pni pclmulqdq ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy cr8_legacy abm sse4a misalig
                         nsse 3dnowprefetch topoext vmmcall fsgsbase bmi1 avx2 smep bmi2 rdseed adx smap clflushopt sha_ni xsaveopt xsavec xgetbv1 clzero xsaveerptr arat npt nrip_save
Virtualization features:
  Hypervisor vendor:     KVM
  Virtualization type:   full
Caches (sum of all):
  L1d:                   32 KiB (1 instance)
  L1i:                   64 KiB (1 instance)
  L2:                    512 KiB (1 instance)
  L3:                    8 MiB (1 instance)
NUMA:
  NUMA node(s):          1
  NUMA node0 CPU(s):     0,1
Vulnerabilities:
  Itlb multihit:         Not affected
  L1tf:                  Not affected
  Mds:                   Not affected
  Meltdown:              Not affected
  Mmio stale data:       Not affected
  Retbleed:              Mitigation; untrained return thunk; SMT vulnerable
  Spec store bypass:     Vulnerable
  Spectre v1:            Mitigation; usercopy/swapgs barriers and __user pointer sanitization
  Spectre v2:            Mitigation; Retpolines, STIBP disabled, RSB filling, PBRSB-eIBRS Not affected
  Srbds:                 Not affected
  Tsx async abort:       Not affected
```
- Manager version
```
# /var/ossec/bin/wazuh-control info
WAZUH_VERSION="v4.6.0"
WAZUH_REVISION="40600"
WAZUH_TYPE="server"
```
- Windows Agent version
![Screenshot 2023-09-15 at 15 48 09](https://github.com/wazuh/wazuh/assets/39094716/31fec2a5-5937-4044-b2b6-d3132b50fced)
- Windows Agent initial config
```xml
<!--
  Wazuh - Agent - Default configuration for Windows
  More info at: https://documentation.wazuh.com
  Mailing list: https://groups.google.com/forum/#!forum/wazuh
-->

<ossec_config>

  <client>
    <server>
      <address>MANAGER_IP</address>
      <port>1514</port>
      <protocol>tcp</protocol>
    </server>
    <crypto_method>aes</crypto_method>
    <notify_time>10</notify_time>
    <time-reconnect>60</time-reconnect>
    <auto_restart>yes</auto_restart>
  </client>

  <!-- Agent buffer options -->
  <client_buffer>
    <disabled>no</disabled>
    <queue_size>5000</queue_size>
    <events_per_second>500</events_per_second>
  </client_buffer>

  <!-- Log analysis -->
  <localfile>
    <location>Application</location>
    <log_format>eventchannel</log_format>
  </localfile>

  <localfile>
    <location>Security</location>
    <log_format>eventchannel</log_format>
    <query>Event/System[EventID != 5145 and EventID != 5156 and EventID != 5447 and
      EventID != 4656 and EventID != 4658 and EventID != 4663 and EventID != 4660 and
      EventID != 4670 and EventID != 4690 and EventID != 4703 and EventID != 4907 and
      EventID != 5152 and EventID != 5157]</query>
  </localfile>

  <localfile>
    <location>System</location>
    <log_format>eventchannel</log_format>
  </localfile>

  <localfile>
    <location>active-response\active-responses.log</location>
    <log_format>syslog</log_format>
  </localfile>

  <!-- Policy monitoring -->
  <rootcheck>
    <disabled>no</disabled>
    <windows_apps>./shared/win_applications_rcl.txt</windows_apps>
    <windows_malware>./shared/win_malware_rcl.txt</windows_malware>
  </rootcheck>

  <!-- Security Configuration Assessment -->
  <sca>
    <enabled>yes</enabled>
    <scan_on_start>yes</scan_on_start>
    <interval>12h</interval>
    <skip_nfs>yes</skip_nfs>
  </sca>

  <!-- File integrity monitoring -->
  <syscheck>

    <disabled>no</disabled>

    <!-- Frequency that syscheck is executed default every 12 hours -->
    <frequency>43200</frequency>

    <!-- Default files to be monitored. -->
    <directories recursion_level="0" restrict="regedit.exe$|system.ini$|win.ini$">%WINDIR%</directories>

    <directories recursion_level="0" restrict="at.exe$|attrib.exe$|cacls.exe$|cmd.exe$|eventcreate.exe$|ftp.exe$|lsass.exe$|net.exe$|net1.exe$|netsh.exe$|reg.exe$|regedt32.exe|regsvr32.exe|runas.exe|sc.exe|schtasks.exe|sethc.exe|subst.exe$">%WINDIR%\SysNative</directories>
    <directories recursion_level="0">%WINDIR%\SysNative\drivers\etc</directories>
    <directories recursion_level="0" restrict="WMIC.exe$">%WINDIR%\SysNative\wbem</directories>
    <directories recursion_level="0" restrict="powershell.exe$">%WINDIR%\SysNative\WindowsPowerShell\v1.0</directories>
    <directories recursion_level="0" restrict="winrm.vbs$">%WINDIR%\SysNative</directories>

    <!-- 32-bit programs. -->
    <directories recursion_level="0" restrict="at.exe$|attrib.exe$|cacls.exe$|cmd.exe$|eventcreate.exe$|ftp.exe$|lsass.exe$|net.exe$|net1.exe$|netsh.exe$|reg.exe$|regedit.exe$|regedt32.exe$|regsvr32.exe$|runas.exe$|sc.exe$|schtasks.exe$|sethc.exe$|subst.exe$">%WINDIR%\System32</directories>
    <directories recursion_level="0">%WINDIR%\System32\drivers\etc</directories>
    <directories recursion_level="0" restrict="WMIC.exe$">%WINDIR%\System32\wbem</directories>
    <directories recursion_level="0" restrict="powershell.exe$">%WINDIR%\System32\WindowsPowerShell\v1.0</directories>
    <directories recursion_level="0" restrict="winrm.vbs$">%WINDIR%\System32</directories>

    <directories realtime="yes">%PROGRAMDATA%\Microsoft\Windows\Start Menu\Programs\Startup</directories>

    <ignore>%PROGRAMDATA%\Microsoft\Windows\Start Menu\Programs\Startup\desktop.ini</ignore>

    <ignore type="sregex">.log$|.htm$|.jpg$|.png$|.chm$|.pnf$|.evtx$</ignore>

    <!-- Windows registry entries to monitor. -->
    <windows_registry>HKEY_LOCAL_MACHINE\Software\Classes\batfile</windows_registry>
    <windows_registry>HKEY_LOCAL_MACHINE\Software\Classes\cmdfile</windows_registry>
    <windows_registry>HKEY_LOCAL_MACHINE\Software\Classes\comfile</windows_registry>
    <windows_registry>HKEY_LOCAL_MACHINE\Software\Classes\exefile</windows_registry>
    <windows_registry>HKEY_LOCAL_MACHINE\Software\Classes\piffile</windows_registry>
    <windows_registry>HKEY_LOCAL_MACHINE\Software\Classes\AllFilesystemObjects</windows_registry>
    <windows_registry>HKEY_LOCAL_MACHINE\Software\Classes\Directory</windows_registry>
    <windows_registry>HKEY_LOCAL_MACHINE\Software\Classes\Folder</windows_registry>
    <windows_registry arch="both">HKEY_LOCAL_MACHINE\Software\Classes\Protocols</windows_registry>
    <windows_registry arch="both">HKEY_LOCAL_MACHINE\Software\Policies</windows_registry>
    <windows_registry>HKEY_LOCAL_MACHINE\Security</windows_registry>
    <windows_registry arch="both">HKEY_LOCAL_MACHINE\Software\Microsoft\Internet Explorer</windows_registry>

    <windows_registry>HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services</windows_registry>
    <windows_registry>HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\KnownDLLs</windows_registry>
    <windows_registry>HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\SecurePipeServers\winreg</windows_registry>

    <windows_registry arch="both">HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Run</windows_registry>
    <windows_registry arch="both">HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\RunOnce</windows_registry>
    <windows_registry>HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\RunOnceEx</windows_registry>
    <windows_registry arch="both">HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\URL</windows_registry>
    <windows_registry arch="both">HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Policies</windows_registry>
    <windows_registry arch="both">HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Windows</windows_registry>
    <windows_registry arch="both">HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Winlogon</windows_registry>

    <windows_registry arch="both">HKEY_LOCAL_MACHINE\Software\Microsoft\Active Setup\Installed Components</windows_registry>

    <!-- Windows registry entries to ignore. -->
    <registry_ignore>HKEY_LOCAL_MACHINE\Security\Policy\Secrets</registry_ignore>
    <registry_ignore>HKEY_LOCAL_MACHINE\Security\SAM\Domains\Account\Users</registry_ignore>
    <registry_ignore type="sregex">\Enum$</registry_ignore>
    <registry_ignore>HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\MpsSvc\Parameters\AppCs</registry_ignore>
    <registry_ignore>HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\MpsSvc\Parameters\PortKeywords\DHCP</registry_ignore>
    <registry_ignore>HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\MpsSvc\Parameters\PortKeywords\IPTLSIn</registry_ignore>
    <registry_ignore>HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\MpsSvc\Parameters\PortKeywords\IPTLSOut</registry_ignore>
    <registry_ignore>HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\MpsSvc\Parameters\PortKeywords\RPC-EPMap</registry_ignore>
    <registry_ignore>HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\MpsSvc\Parameters\PortKeywords\Teredo</registry_ignore>
    <registry_ignore>HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\PolicyAgent\Parameters\Cache</registry_ignore>
    <registry_ignore>HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\RunOnceEx</registry_ignore>
    <registry_ignore>HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\ADOVMPPackage\Final</registry_ignore>

    <!-- Frequency for ACL checking (seconds) -->
    <windows_audit_interval>60</windows_audit_interval>

    <!-- Nice value for Syscheck module -->
    <process_priority>10</process_priority>

    <!-- Maximum output throughput -->
    <max_eps>100</max_eps>

    <!-- Database synchronization settings -->
    <synchronization>
      <enabled>yes</enabled>
      <interval>5m</interval>
      <max_eps>10</max_eps>
    </synchronization>
  </syscheck>

  <!-- System inventory -->
  <wodle name="syscollector">
    <disabled>no</disabled>
    <interval>1h</interval>
    <scan_on_start>yes</scan_on_start>
    <hardware>yes</hardware>
    <os>yes</os>
    <network>yes</network>
    <packages>yes</packages>
    <ports all="no">yes</ports>
    <processes>yes</processes>

    <!-- Database synchronization settings -->
    <synchronization>
      <max_eps>10</max_eps>
    </synchronization>
  </wodle>

  <!-- CIS policies evaluation -->
  <wodle name="cis-cat">
    <disabled>yes</disabled>
    <timeout>1800</timeout>
    <interval>1d</interval>
    <scan-on-start>yes</scan-on-start>

    <java_path>\\server\jre\bin\java.exe</java_path>
    <ciscat_path>C:\cis-cat</ciscat_path>
  </wodle>

  <!-- Osquery integration -->
  <wodle name="osquery">
    <disabled>yes</disabled>
    <run_daemon>yes</run_daemon>
    <bin_path>C:\Program Files\osquery\osqueryd</bin_path>
    <log_path>C:\Program Files\osquery\log\osqueryd.results.log</log_path>
    <config_path>C:\Program Files\osquery\osquery.conf</config_path>
    <add_labels>yes</add_labels>
  </wodle>

  <!-- Active response -->
  <active-response>
    <disabled>no</disabled>
    <ca_store>wpk_root.pem</ca_store>
    <ca_verification>yes</ca_verification>
  </active-response>

  <!-- Choose between plain or json format (or both) for internal logs -->
  <logging>
    <log_format>plain</log_format>
  </logging>

</ossec_config>

<!-- END of Default Configuration. -->
```
- Windows Agent: Internal configuration
```
# local_internal_options.conf
#
# This file should be handled with care. It contains
# run time modifications that can affect the use
# of OSSEC. Only change it if you know what you
# are doing. Look first at ossec.conf
# for most of the things you want to change.
#
# This file will not be overwritten during upgrades
# but will be removed when the agent is un-installed.
windows.debug=2
```
- Agents connected to the Manager
```
# /var/ossec/bin/agent_control -l

Wazuh agent_control. List of available agents:
   ID: 000, Name: MANAGER_IP (server), IP: 127.0.0.1, Active/Local
   ID: 001, Name: NAME, IP: any, Active

List of agentless devices:
```
- Path from where the tests are being executed (Windows agent)
```
$ pwd
/c/Users/Administrator/wazuh-qa
```
- Running the tests (stopping at the first failure with `-x` and enable debugging with `--pdb`
```python
$ python -m pytest -vvv -x tests/integration/test_syscollector/ --html="report-1.html" --pdb

tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_deactivation[check_syscollector_deactivation] PASSED [  3%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_all_scans_disabled[all_scans_disabled] PASSED [  7%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_disabled] PASSED [ 11%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_interval] PASSED [ 14%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_scan_on_start] PASSED [ 18%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_hardware] PASSED [ 22%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_os] PASSED [ 25%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_network] PASSED [ 29%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_packages] PASSED [ 33%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_ports] PASSED [ 37%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_all_ports] PASSED [ 40%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_processes] PASSED [ 44%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_disabled] PASSED [ 48%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_interval] PASSED [ 51%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_scan_on_start] PASSED [ 55%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_hardware] PASSED [ 59%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_os] PASSED [ 62%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_network] FAILED [ 66%]
.
.
.
>           evm.check_tag_error(file_monitor=file_monitor, field=field)

test_syscollector\test_syscollector_configuration.py:300: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\Python37\lib\site-packages\wazuh_testing-4.6.0-py3.7.egg\wazuh_testing\modules\syscollector\event_monitor.py:299: in check_tag_error
    file_monitor=file_monitor)
C:\Python37\lib\site-packages\wazuh_testing-4.6.0-py3.7.egg\wazuh_testing\event_monitor.py:41: in check_event
    callback=make_callback(callback, prefix), error_message=error_message)
C:\Python37\lib\site-packages\wazuh_testing-4.6.0-py3.7.egg\wazuh_testing\tools\monitoring.py:204: in start
    error_message=error_message).result()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <wazuh_testing.tools.monitoring.QueueMonitor object at 0x000001969DF855C8>
timeout = 5
callback = <function make_callback.<locals>.<lambda> at 0x000001969E0053A8>
accum_results = 1, update_position = True, timeout_extra = 0
error_message = "Could not find this event in None: ERROR: Invalid content for tag 'network' at module 'syscollector'."

    def start(self, timeout=-1, callback=_callback_default, accum_results=1, update_position=True, timeout_extra=0,
              error_message=''):
        """Start the queue monitoring until the stop method is called."""
        if not self._continue:
            self._continue = True
            self._abort = False
            result = None
    
            while self._continue:
                if self._abort:
                    self.stop()
                    if error_message:
                        logger.error(error_message)
                        logger.error(f"Results accumulated: "
                                     f"{len(result) if isinstance(result, list) else 0}")
                        logger.error(f"Results expected: {accum_results}")
>                   raise TimeoutError(error_message)
E                   TimeoutError: Could not find this event in None: ERROR: Invalid content for tag 'network' at module 'syscollector'.

C:\Python37\lib\site-packages\wazuh_testing-4.6.0-py3.7.egg\wazuh_testing\tools\monitoring.py:470: TimeoutError
```

</details>

***

### Proposed fix

This is the fix proposed for the test:

```python
# tests/integration/test_syscollector/test_syscollector_configuration.py
.
.
.
import os
import sys
import time
.
.
.
def test_syscollector_invalid_configurations(...):
    .
    .
    .
    # Wait 100 ms to avoid race condition
    time.sleep(0.1)
```


After executing the tests with said fix they've passed successfully:

```shell
$ python -m pytest -vvv -x tests/integration/test_syscollector/ --html="report-1.html" --pdb -k "not test_syscollector_scanning"
============================= test session starts =============================
platform win32 -- Python 3.8.0, pytest-6.2.2, py-1.10.0, pluggy-0.13.1 -- C:\Users\Administrator\AppData\Local\Programs\Python\Python38\python.exe
cachedir: .pytest_cache
metadata: {'Python': '3.8.0', 'Platform': 'Windows-10-10.0.20348-SP0', 'Packages': {'pytest': '6.2.2', 'pluggy': '0.13.1'}, 'Plugins': {'html': '3.1.1', 'metadata': '2.0.4', 'testinfra': '5.0.0'}}
rootdir: C:\Users\Administrator\wazuh-qa\tests\integration, configfile: pytest.ini
plugins: html-3.1.1, metadata-2.0.4, testinfra-5.0.0
collecting ... collected 27 items / 1 deselected / 26 selected

tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_deactivation[check_syscollector_deactivation] PASSED [  3%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_all_scans_disabled[all_scans_disabled] PASSED [  7%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_disabled] PASSED [ 11%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_interval] PASSED [ 15%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_scan_on_start] PASSED [ 19%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_hardware] PASSED [ 23%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_os] PASSED [ 26%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_network] PASSED [ 30%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_packages] PASSED [ 34%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_ports] PASSED [ 38%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_all_ports] PASSED [ 42%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_processes] PASSED [ 46%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_disabled] PASSED [ 50%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_interval] PASSED [ 53%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_scan_on_start] PASSED [ 57%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_hardware] PASSED [ 61%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_os] PASSED [ 65%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_network] PASSED [ 69%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_packages] PASSED [ 73%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_ports] PASSED [ 76%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_all_ports] PASSED [ 80%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_processes] PASSED [ 84%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_max_eps] PASSED [ 88%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[empty_value_hotfixes] PASSED [ 92%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_invalid_configurations[invalid_value_hotfixes] PASSED [ 96%]
tests\integration\test_syscollector\test_syscollector_configuration.py::test_syscollector_default_values[check_default_values] XFAIL [100%]

-- generated html file: file://C:\Users\Administrator\wazuh-qa\report-1.html --
=========== 25 passed, 1 deselected, 1 xfailed in 232.47s (0:03:52) ===========
```

**Comment by Dwordcito** [comment]
It is not a product issue according to the creator of the issue.

This type of failure is because the database file is taken by another process.
If you notice this problem under unwanted behavior, please reopen it.

**Comment by GabrielEValenzuela** [comment]
Update Report
---------------------


We found a possible relation with a previous issue that shows the same behavior in the FIM database. Issue #4692 and its corresponding pull request #4708 introduce a mechanism to retry the deletion of the database file due to a failure with the host operating system, which does not release the owner of the file immediately and causes the error of the `std::remove` function. We tested this behavior in our environment and we're working in a fix.
